test/Microsoft.NET.TestFramework/Commands/PackCommand.cs

```diff
@@ -6,15 +6,13 @@ namespace Microsoft.NET.TestFramework.Commands
     public sealed class PackCommand : MSBuildCommand
     {
         public PackCommand(ITestOutputHelper log, string projectPath, string? relativePathToProject = null)
-            : base(log, "Pack", projectPath, relativePathToProject)
```

Comment:
There are [numerous projects on github that appear to use both `PackAsTool` and `GeneratePackageOnBuild`](https://github.com/search?q=GeneratePackageOnBuild+PackAsTool&type=code).

From just a quick look at this PR, it's unclear to me whether these changes will cause a problem for this scenario.